### ISSUE TITLE
Floating point bug in ineptpdf.py

### ISSUE THREAD

**Issue by Difflugia** [apprenticeharper/DeDRM_tools#251]
Under certain circumstances, tiny floating point numbers from an input PDF are rendered in scientific notation in the output PDF, which are not handled gracefully by Acrobat.

I fixed this by replacing lines 2145-2148:

```
        elif isinstance(obj, (int, long, float)):
            if self.last.isalnum():
                self.write(' ')
            self.write(str(obj))
```
with

```
       elif isinstance(obj, (int, long)):
            if self.last.isalnum():
                self.write(' ')
            self.write(str(obj))
        elif isinstance(obj, float):
            if self.last.isalnum():
                self.write(' ')
            self.write('%0.15f' % obj)
```

This is probably not the best solution, either, however. Since one of the source files I tested had floats with over 20 decimal places (which Python didn't handle), the output didn't exactly match the input. The module should probably be modified to ensure that any float numbers in the source are represented identically in the output.

Since I'm not a Python programmer, I figured I'd leave this to someone else. Unless I'm the only one left that DeDRMs PDFs, I guess.

**Comment by apprenticeharper** [comment]
Thank you for this bug report and diagnosis. I'm not sure of the best way to fix this either. Pull requests welcome from anyone with a good idea!

**Comment by apprenticeharper** [comment]
I've taken a look, and it seems to me that use of the Decimal module instead of float would be the ideal solution. I just need to find a free but DRMed PDF to test this on!

**Comment by apprenticeharper** [comment]
I have failed to find an Adobe DRMed PDF to test. If someone has a small-ish Adobe DRM PDF, and is willing to send it to me along with their Adobe key file, it would be much appreciated. apprenticeharper@gmail.com

**Comment by ElleKayEm** [comment]
Here's a link to a DRMed freebie on Google Play that's available in both PDF and EPUB:

https://play.google.com/store/books/details?id=TeAmICbS-QIC

**Comment by apprenticeharper** [comment]
Unfortunately, Google Play insists on a payment method being preset in an account before allowing free books to be 'purchased'.

**Comment by ElleKayEm** [comment]
Hmm, I "bought" it and was going to send it to you.  But it's not showing up in my books for me to download it.  If it shows, I'll send it or look for something else.

**Comment by ElleKayEm** [comment]
Found another book I already had and sent it to you.

**Comment by apprenticeharper** [comment]
Thanks for the sample book.

This bug should be fixed in 6.5.4. (I switched to using Decimal instead of float, which should preserve all the precision of the original document.)

A confirmation of the fix would be appreciated.

**Comment by apprenticeharper** [comment]
If anyone can confirm the fix in 6.5.4 I'd be grateful.

**Comment by apprenticeharper** [comment]
I'm assuming it's fixed.
